DepotDownloader/AnsiDetector.cs

```diff
@@ -0,0 +1,133 @@
+// Copied from https://github.com/spectreconsole/spectre.console/blob/d79e6adc5f8e637fb35c88f987023ffda6707243/src/Spectre.Console/Internal/Backends/Ansi/AnsiDetector.cs
+// which is based on https://github.com/keqingrong/supports-ansi/blob/master/index.js
+// <auto-generated/>
+
+using System;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text.RegularExpressions;
+
+namespace Spectre.Console;
```

Comment:
Should we just use/ref their NuGet package?

Reply:
I wanted to use their package for displaying a fancier progress, but the downloader code got refactored that its not very realisting to get per file progress now, don't think it's worth pulling the whole pkg for it.